File: python/account.py
class NoMoneyError(BaseException):
    pass

class Account:
    def __init__(self, name):
        self.money = 0
        self.name = name
    
    def deposit(self, amount):
        self.money += amount
    
    def spend(self, amount):
        if self.money < amount:
            print("돈이 없습니다")
            raise NoMoneyError()
        self.money -= amount

    def showMoneny(self):
        print(self.money)
    
    def sendTo(self, target, amount):
        try:
            self.spend(amount)
            target.deposit(amount)
        except:
            pass
        
    def __repr__(self):
        return "Account(" + self.name + ") left = " + str(self.money)

    def __str__(self):
        return "str version of " + self.__repr__()

account1 = Account("계좌 1")
account2 = Account("받을 계좌")
account1.deposit(10000)
account1.showMoneny()
account1.sendTo(account2, 20000)

File: java/java-intellij/account/src/com/naver/account/Account.java
package com.naver.account;

public class Account {
    private String name;
    private String accountNumber;
    private long money;

    int draw(int amount) {
        if (amount < 0) {
            System.out.println("음수를 인출하려 했습니다.");
            return 0;
        }
        if (amount > money) {
            System.out.println(name + "가 잔액 " + money + "보다 많은 돈" + amount + "를 인출하려 했습니다.");
            return 0;
        }
        money -= amount;
        return amount;
    }

    int deposit(int amount) {
        if (amount < 0) {
            System.out.println("음수를 입금하려 했습니다.");
            return 0;
        }
        money += amount;
        return amount;
    }

    int send(int amount, Account target) {
        if(draw(amount) == 0) {
            System.out.println("송금 실패");
            return 0;
        }
        target.deposit(amount);
        return amount;
    }

    //    생성자(Constructor). 객체의 초기화를 담당
    private Account(String name, String accountNumber, long initMoney) {
        this.name = name;
        this.accountNumber = accountNumber;
        this.money = initMoney;
    }

    void describe() {
        System.out.println(name + "의 계좌 정보=============");
        System.out.println("계좌번호:" + accountNumber);
        System.out.println("잔액:" + money);
    }

    static Account createAccount(String name, String accountNumber, long initMoney) {
        System.out.println("계좌생성");
        return new Account(name, accountNumber, initMoney);
    }
}
